frontend/src/lib/components/popups/ChartConfigPopup.svelte

```diff
@@ -0,0 +1,63 @@
+<script lang="ts">
+	import type { ChartConfig, ZenoService } from '$lib/zenoapi';
+	import Button from '@smui/button';
+	import { Content } from '@smui/paper';
+	import Textfield from '@smui/textfield';
+	import { createEventDispatcher, getContext } from 'svelte';
+	import Popup from './Popup.svelte';
+
+	export let config: ChartConfig;
+	export let chartId: number | null = null;
+
+	const oldConfig = { ...config };
+	const dispatch = createEventDispatcher();
+	const zenoClient = getContext('zenoClient') as ZenoService;
+
+	function updateChartConfig() {
+		zenoClient.updateChartConfig(config, chartId).then(() => {
+			dispatch('close');
+		});
+	}
+
+	function undoChartConfig() {
+		config = oldConfig;
+		dispatch('close');
+	}
+
+	function resetChartConfig() {
+		zenoClient.deleteChartConfig(config.projectUuid, chartId).then(() => {
```

Comment:
This needs a higher z-index I think, the vega settings circle is shown on top of it.